src/rgw/rgw_lua_utils.cc

```diff
@@ -160,16 +156,16 @@ lua_state_guard::~lua_state_guard() {
   if (!L) {
     return;
   }
-  void* ud = nullptr;
-  lua_getallocf(L, &ud);
-  auto remaining_memory = static_cast<std::size_t*>(ud);
-
-  if (remaining_memory) {
-    const auto used_memory = max_memory - *remaining_memory;
-    ldpp_dout(dpp, 20) << "Lua is using: " << used_memory << 
-      " bytes (" << 100.0*used_memory/max_memory << "%)" << dendl;
+  int temp_max_memory = max_memory;
+  if (mem_in_use) {
```

Comment:
nit:
```suggestion
  if (max_memory > 0) {
```
you can print `0%` memory in use, if there is not memory used on dtor.
also, this will simplify the `ldpp_dout` expression